File: solutions/Question 2114/Solution4.py
# Problem: 2114. Maximum Number of Words Found in Sentences
# Difficulty: Easy
# Topics: Array, String
# URL: https://leetcode.com/problems/maximum-number-of-words-found-in-sentences/
# Runtime: 3 ms
# Memory: 18 MB
# Solved via CodePath Auto-Committer

class Solution:
    def mostWordsFound(self, sentences: List[str]) -> int:
        # a = 0
        # for b in sentences:
            # x = len(b.split(" "))
            # if x > a:
                # a = x
        # return a
        return max(len(x.split()) for x in sentences)

File: solutions/Question 1672/Solution4.cpp
/*
 * Problem: 1672. Richest Customer Wealth
 * Difficulty: Easy
 * Topics: Array, Matrix
 * URL: https://leetcode.com/problems/richest-customer-wealth/
 * Runtime: 0 ms
 * Memory: 11.4 MB
 * Solved via CodePath Auto-Committer
 */

class Solution {
public:
    int maximumWealth(vector<vector<int>>& accounts) {
        int richest= 0;
        for (const auto& row : accounts) {
            int sum = accumulate(row.begin(), row.end(), 0) ;
            if (sum >= richest) richest = sum;
        }
        return richest;

        
    }
};